src/twisted/web/static.py

```diff
@@ -1057,13 +1065,11 @@ def render(self, request):
 
         tableContent = "".join(self._buildTableContent(dirs + files))
 
-        header = "Directory listing for %s" % (
-            escape(unquote(nativeString(request.uri))),)
+        header = u"Directory listing for {}".format(
+            escape(unquote(request.uri.decode("utf-8"))))
 
         done = self.template % {"header": header, "tableContent": tableContent}
-        if _PY3:
-            done = done.encode("utf8")
-
+        done = done.encode("utf-8")
```

Comment:
This can cause an encoding error with UTF-8 on Python 2, because `done` will first be decoded as ASCII.  The `_PY3` check should remain.

Reply:
No, this is always needed on Python 2 and 3.  Now, done is always unicode, due to header being unicode,
but bytes must be returned from this function.